R/zzz.R

```diff
@@ -28,133 +27,30 @@
-        } else {
-          kwargs[[valuenames[[i]]]] <- values[[i]]
-        }
-      }
-    }
-  }
-  list(args = args, kwargs = kwargs)
-}
-
-.cleanUpStackTrace <- function(callable, args) {
-  conn <- textConnection("outputCapture", open = "w", local = TRUE)
-  sink(conn)
-  tryCatch(
-    {
-      result <- do.call(callable, args)
-      sink()
-      close(conn)
-      cat(paste(outputCapture, collapse = ""))
-      result
-    },
-    error = function(e) {
-      sink()
-      close(conn)
-      errorToReport <- paste(c(outputCapture, e$message), collapse = "\n")
-      if (!getOption("verbose")) {
-        # extract the error message
-        splitArray <- strsplit(errorToReport, "exception-message-boundary", fixed = TRUE)[[1]]
-        if (length(splitArray) >= 2) errorToReport <- splitArray[2]
-      }
-      stop(errorToReport)
-    }
-  )
-}
-
-.defineFunction <- function(synName, pyName, functionContainerName) {
-  force(synName)
-  force(pyName)
-  force(functionContainerName)
-  assign(sprintf(".%s", synName), function(...) {
-    functionContainer <- pyGet(functionContainerName, simplify = FALSE)
-    argsAndKwArgs <- .determineArgsAndKwArgs(...)
-    functionAndArgs <- append(list(functionContainer, pyName), argsAndKwArgs$args)
-    returnedObject <- .cleanUpStackTrace(
-      pyCall,
-      list(
-        "gateway.invoke",
-        args = functionAndArgs,
-        kwargs = argsAndKwArgs$kwargs,
-        simplify = F
-      )
-    )
-    .objectDefinitionHelper(returnedObject)
-  })
-  setGeneric(
-    name = synName,
-    def = function(...) {
-      do.call(sprintf(".%s", synName), args = list(...))
-    }
-  )
-}
-
-.defineConstructor <- function(synName, pyName) {
-  force(synName)
-  force(pyName)
-  assign(sprintf(".%s", synName), function(...) {
-    synapseClientModule <- pyGet("synapseclient")
-    argsAndKwArgs <- .determineArgsAndKwArgs(...)
-    functionAndArgs <- append(
-      list(synapseClientModule, pyName),
-      argsAndKwArgs$args
-    )
-    .cleanUpStackTrace(
-      pyCall,
-      list(
-        "gateway.invoke",
-        args = functionAndArgs,
-        kwargs = argsAndKwArgs$kwargs,
-        simplify = F
-      )
-    )
-  })
-  setGeneric(
-    name = synName,
-    def = function(...) {
-      do.call(sprintf(".%s", synName), args = list(...))
-    }
-  )
+.callback <- function(name, def) {
+  setGeneric(name, def)
 }
 
 .defineRPackageFunctions <- function() {
-  functionInfo <- .getSynapseFunctionInfo(system.file(package = "synapser"))
-  for (f in functionInfo) {
-    .defineFunction(f$synName, f$name, f$functionContainerName)
-  }
-  classInfo <- .getSynapseClassInfo(system.file(package = "synapser"))
-  for (c in classInfo) {
-    .defineConstructor(c$name, c$name)
-  }
+  generateRWrappers(pyPkg = "synapseclient",
+                    container = "synapseclient.Synapse",
+                    setGenericCallback = .callback,
+                    functionFilter = synapseFunctionSelector,
```

Comment:
may have to add "." prefix to functions like synapseFunctionSelector to keep them out of the package namespace.